src/riak_kv_pb_timeseries.erl

```diff
@@ -362,6 +374,41 @@ row_to_key(Row, DDL, Mod) ->
     riak_kv_ts_util:encode_typeval_key(
       riak_ql_ddl:get_partition_key(DDL, Row, Mod)).
 
+%%%%%%%%
+%% Utility functions for batch delivery of records
+termsize(Term) ->
+    size(term_to_binary(Term)).
+
+pick_batch_option(_, _, Records, _, 1) ->
+    {individual, Records};
+pick_batch_option(true, MaxBatch, Records, SampleSize, _NumRecs) ->
+    {batches, create_batches(Records,
+                             estimated_row_count(SampleSize, MaxBatch))};
+pick_batch_option(false, _, Records, _, _) ->
+    {individual, Records}.
+
+estimated_row_count(SampleRowSize, MaxBatchSize) ->
+    %% Assume some rows will be larger, so introduce a fudge factor of
+    %% roughly 10 percent.
+    RowSizeFudged = (SampleRowSize * 10) div 9,
+    MaxBatchSize div RowSizeFudged.
+
+create_batches(Rows, MaxSize) when length(Rows) =< MaxSize ->
+    [Rows];
+%% May be a more efficient way to do this. Take a list of arbitrary
+%% data (expected to be a list of lists for this use case) and create
+%% a list of MaxSize lists.
+create_batches(Rows, MaxSize) ->
+    create_batches(Rows, MaxSize, MaxSize, [], []).
+
+create_batches([], _Counter, _Max, ThisBatch, AllBatches) ->
+    AllBatches ++ [ThisBatch];
```

Comment:
My expectation is that this will be invoked rarely, and I thought the VM had optimized this idiom pretty well, but no, I don't think order matters. Even if all batches are retained in the correct order, there's no guarantee they'll arrive at the remote vnode (and hence the backend) in that same order.

Willing to change it, unconvinced it's worth doing.